File: java/ds-recruit-java/src/main/java/org/recruit/utils/MyGreeter.java
package org.recruit.utils;

import org.springframework.stereotype.Component;

import java.util.Calendar;
import java.util.Date;

@Component
public class MyGreeter {
    public String greeting() {
        Calendar calendar = Calendar.getInstance();
        calendar.setTime(new Date());
        int hour = calendar.get(Calendar.HOUR_OF_DAY);

        if (hour >= 6 && hour < 12) {
            return "Good morning";
        } else if (hour >= 12 && hour < 18) {
            return "Good afternoon";
        } else {
            return "Good evening";
        }
    }
}

File: php/Src/MyGreeter.php
<?php

namespace Src;

class MyGreeter
{

    /**
     * return different value according to hour
     * @return string
     * 6-12 Good morning
     * 12-18 Good afternoon
     * else Good evening
     */
    public function greeting():string{
        $currentHour = date('H');
        // return different string depending on hour
        if ($currentHour >= 6 && $currentHour < 12) {
            return "Good morning";
        } elseif ($currentHour >= 12 && $currentHour < 18) {
            return "Good afternoon";
        } else {
            return "Good evening";
        }
    }

}